src/vma/dev/ring_bond.h

```diff
@@ -75,12 +73,13 @@ class ring_bond : public ring {
 #ifdef DEFINED_SOCKETXTREME		
 	int 			socketxtreme_poll(struct vma_completion_t *vma_completions, unsigned int ncompletions, int flags);
 #endif // DEFINED_SOCKETXTREME		
+	virtual void    slave_create(int if_index) = 0;
+	virtual void    slave_destroy(int if_index) = 0;
 protected:
-	virtual void		create_slave_list(in_addr_t local_if, ring_resource_creation_info_t* p_ring_info, bool active_slaves[], uint16_t partition) = 0;
 	void			update_rx_channel_fds();
-	void			close_gaps_active_rings();
-	ring_simple**		m_bond_rings;
-	ring_simple**		m_active_rings;
+	void			popup_active_rings();
+	ring_slave_vector_t     m_bond_rings;
+	lock_mutex              m_lock;
```

Comment:
Why we need 3 locks here ? m_lock, m_lock_ring_rx, m_lock_ring_tx ? why mutex ? please add info about the lock usage